internal/check/handler.go

```diff
@@ -126,12 +140,19 @@ func (h *Handler) getCheck(w http.ResponseWriter, r *http.Request, _ httprouter.
 //       403: getCheckResponse
 //       500: genericError
 func (h *Handler) postCheck(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	maxDepth, err := x.GetMaxDepthFromQuery(r.URL.Query(), false)
```

Comment:
Hm, just thinking about whether this endpoint, as it reads all other values from the JSON body, should also consume the `max-depth` from the body. What do you think about this inconsistency? On the other hand, `max-depth` is a kind of meta-parameter, it will never contain sensitive information.

Reply:
I thought to put the parameter in the JSON body initially but I think a query param is fine. It definitely is a meta-parameter, and perhaps the right place for those is in the query params.